### PR TITLE
Removes foreign keys

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/7682

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- install version 1.0 and populate data (visit sited, do tips, set recurring tips, etc)
- update to this version
- make sure that all data is still there

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by NejcZdovc** [comment]
I tried to limit it, so rest will be done in https://github.com/brave/brave-browser/issues/7145

File: components/brave_rewards/browser/database/database_activity_info.cc [hunk 1]
```diff
@@ -443,6 +447,46 @@ bool DatabaseActivityInfo::MigrateToV6(sql::Database* db) {
   return true;
 }
 
+bool DatabaseActivityInfo::MigrateToV15(sql::Database* db) {
```
Comment (tmancey) [inline components/brave_rewards/browser/database/database_activity_info.cc]: Checks if db is nullptr/not reached

File: components/brave_rewards/browser/database/database_contribution_info_publishers.cc [hunk 2]
```diff
@@ -13,6 +13,10 @@
 #include "sql/statement.h"
 #include "sql/transaction.h"
 
+namespace {
+  const char table_name_[] = "contribution_info_publishers";
```
Comment (tmancey) [inline components/brave_rewards/browser/database/database_contribution_info_publishers.cc]: Table_name_ is not a member variable _

File: components/brave_rewards/browser/database/database_contribution_queue.h [hunk 3]
```diff
@@ -35,10 +31,14 @@ class DatabaseContributionQueue: public DatabaseTable {
   bool DeleteAllRecords(sql::Database* db);
 
  private:
+  bool CreateTableV9(sql::Database* db);
+
+  bool MigrateToV9(sql::Database* db);
+
+  bool MigrateToV15(sql::Database* db);
+
```
Comment (masparrow) [inline components/brave_rewards/browser/database/database_contribution_queue.h]: nit: observation - as all Create/Migrate methods operate on `sql::Database` all such methods should be const? (As a practice across the board I mean - there's no need to change this PR. But, if agreed we can address in another ticket).

File: components/brave_rewards/browser/database/publisher_info_database.cc [hunk 4]
```diff
@@ -706,6 +675,23 @@ std::string PublisherInfoDatabase::GetSchema() {
 
 // Migration -------------------------------------------------------------------
 
+bool PublisherInfoDatabase::MigrateV0toV1() {
```
Comment (masparrow) [inline components/brave_rewards/browser/database/publisher_info_database.cc]: question: `V0`? What is / Why would version zero exist?

Comment (NejcZdovc) [reply components/brave_rewards/browser/database/publisher_info_database.cc]: version 0 is when database was not created yet, meaning that we need to start from scratch an create version 1

File: components/brave_rewards/browser/database/database_contribution_queue.h (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/database_contribution_queue.h]: sure you can add this to your ticket

Comment (masparrow) [reply components/brave_rewards/browser/database/database_contribution_queue.h]: Done :+1:

File: components/brave_rewards/browser/database/publisher_info_database.cc (hunk 4 above)
Comment (masparrow) [reply components/brave_rewards/browser/database/publisher_info_database.cc]: Ok - was just puzzled that we'd need to migrate if there was no existing DB to migrate from.

**Pull request merged**